pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerStatsRecorderImpl.java

```diff
@@ -196,6 +223,20 @@ public void incrementNumAcksReceived(long latencyNs) {
         }
     }
 
+    @Override
+    public void incrementBatchSize(int batchSize) {
```

Comment:
`incrementBatchSize` and `incrementMsgSize` can be done in `updateNumMsgsSent`, no need for new api.

Reply:
And missed collect from send operation in  `org.apache.pulsar.client.impl.ProducerImpl#recoverProcessOpSendMsgFrom`.